src/libraries/Microsoft.Extensions.Options/src/OptionsFactory.cs

```diff
@@ -80,7 +80,7 @@ public TOptions Create(string name)
                 }
                 if (failures.Count > 0)
                 {
-                    throw new OptionsValidationException(name, typeof(TOptions), failures);
+                    throw new OptionsValidationException(name!, typeof(TOptions), failures);
```

Comment:
It really seems like it should, bet there's an explicit `throw` if it is. This was one of the questions from the PR description:

> In `OptionsFactory.Create` `name` parameter looks like it should be nullable, but `OptionsValidationException` will throw if `name` is `null`

Since you agree that constructor should accept `null`, I've removed `throw` inside it.